src/audio_core/hle/fdk_decoder.cpp

```diff
@@ -186,28 +188,22 @@ std::optional<BinaryResponse> FDKDecoder::Impl::Decode(const BinaryRequest& requ
             return std::nullopt;
         }
     }
+
     // transfer the decoded buffer from vector to the FCRAM
-    if (out_streams[0].size() != 0) {
-        if (request.dst_addr_ch0 < Memory::FCRAM_PADDR ||
-            request.dst_addr_ch0 + out_streams[0].size() >
-                Memory::FCRAM_PADDR + Memory::FCRAM_SIZE) {
-            LOG_ERROR(Audio_DSP, "Got out of bounds dst_addr_ch0 {:08x}", request.dst_addr_ch0);
-            return {};
+    for (auto ch = 0; ch < out_streams.size(); ch++) {
```

Comment:
```suggestion
    for (std::stize_t ch = 0; ch < out_streams.size(); ch++) {
```

`auto` will turn it into `int` and gcc will complain about sign compare